src/base/bittorrent/session.h

```diff
@@ -63,7 +63,16 @@ enum MaxRatioAction
 {
     Pause,
     Remove,
-    EnableSuperSeeding
+    EnableSuperSeeding,
+    DeleteFiles,
+    DeleteFolder
+};
+
+enum DeleteOption
+{
+    Torrent,
+    TorrentAndFiles,
+    TorrentAndFolder
```

Comment:
I think it is more sensible to rename it to `TorrentAndRootFolder`.

Reply:
How should it work for torrent that has no root folder?
I suppose the main purpose of this option is not only remove torrent root folder but remove torrent with all folders belonging to it regardless of the fact that there are some 3rd party files exist in these folders. When you talk about the root folder, this is just a special case of the above behavior, isn't it?

Reply:
Yes you are right, I wasn't aware of the main idea behind it.
However the final name should correlate with how we name it in the UI.

Reply:
If I am correct in my assumption of its purpose, it should be called "Delete torrent and folders" in UI and TorrentAndFolders here.

Reply:
From the other side it deletes not only folders (some files can be outside any folder). Maybe "Delete torrent, files and folders"?

Reply:
>From the other side it deletes not only folders (some files can be outside any folder). Maybe "Delete torrent, files and folders"?

👍  This sounds better than other current alternatives.

Reply:
>"Delete torrent, files and folders"

If "files" are mentioned people will wonder why this option deletes all files while the other only the torrent's files.

Reply:
>why this option deletes all files while the other only the torrent's files.

IMO, this option raised more questions than answers from the very beginning.
Why is it even necessary? Why should qBittorrent delete unrelated files?
It looks like the original author just wanted to inject something subjective into the app's features.
I still think "delete torrent and files" is the only thing that most affected users lack.

Reply:
>Why should qBittorrent delete unrelated files?

IMO we shouldn't care about unrelated files, this way we can maintain less.

>It looks like the original author just wanted to inject something subjective into the app's features.

After re-reading the reason in https://github.com/qbittorrent/qBittorrent/pull/8671#discussion_r178464345 I agree with you. There are other ways to achieve what he wants and doesn't need this option.

>I still think "delete torrent and files" is the only thing that most affected users lack.

Seems so, none of the issues in the opening post mentioned this option.

@thalieht 
I would suggest that we drop this option for now, it will be easier to finish this PR and we can pick it up later if some stronger/more valid arguments comes in.

Reply:
>Why is it even necessary? Why should qBittorrent delete unrelated files?
It looks like the original author just wanted to inject something subjective into the app's features.
I still think "delete torrent and files" is the only thing that most affected users lack.

Personally i don't care about it in the ratio options but if this option was available in regular delete, it would be the only option i would use. There is never a time i want to keep foreign stuff in the torrent folder.

Don't tell me you guys want to drop it only because of the name? If it's also because of `Utils::Fs::removeDirRecursive` not being appropriate for it then ok. In any case here's another suggestion: "Delete torrents and their files" "Delete torrents and their folder".

Reply:
>There is never a time i want to keep foreign stuff in the torrent folder.

Then why do you put "foreign stuff" there?
I can't understand how it can come there under normal use?

Reply:
>Don't tell me you guys want to drop it only because of the name?

For me, that has nothing to do with the main reason (which is mentioned in my first sentence [above](https://github.com/qbittorrent/qBittorrent/pull/11241#discussion_r325485878)).

>If it's also because of Utils::Fs::removeDirRecursive not being appropriate for it then ok.

This is part of the reason (for me), albeit a minor one.

BTW, I also found a bug (unrelated to this PR) that qbt will remove user created empty folders when using `shift + del` to remove torrent which should not happen.

Reply:
Ok i'll drop it.
>Then why do you put "foreign stuff" there?
I can't understand how it can come there under normal use?

Temp stuff e.g. unzipped, subtitles etc.